acceptance/tests/connect/connect_proxy_lifecycle_test.go

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package connect
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul-k8s/acceptance/framework/connhelper"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/consul"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/helpers"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/k8s"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/logger"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/hashicorp/go-version"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type LifecycleShutdownConfig struct {
+	secure     bool
+	helmValues map[string]string
+}
+
+const (
+	helmDrainListenersKey     = "connectInject.sidecarProxy.lifecycle.defaultEnableShutdownDrainListeners"
+	helmGracePeriodSecondsKey = "connectInject.sidecarProxy.lifecycle.defaultShutdownGracePeriodSeconds"
+)
+
+// Test the endpoints controller cleans up force-killed pods.
+func TestConnectInject_ProxyLifecycleShutdown(t *testing.T) {
+	cfg := suite.Config()
+
+	ver, err := version.NewVersion("1.2.0")
+	require.NoError(t, err)
+	if cfg.ConsulDataplaneVersion != nil && cfg.ConsulDataplaneVersion.LessThan(ver) {
+		t.Skipf("skipping this test because proxy lifecycle management is not supported in consul-dataplane version %v", cfg.ConsulDataplaneVersion.String())
+	}
+
+	for _, testCfg := range []LifecycleShutdownConfig{
+		{secure: false, helmValues: map[string]string{}},
+		{secure: true, helmValues: map[string]string{}},
+		{secure: false, helmValues: map[string]string{
+			helmDrainListenersKey:     "true",
+			helmGracePeriodSecondsKey: "30",
+		}},
+		{secure: true, helmValues: map[string]string{
+			helmDrainListenersKey:     "true",
+			helmGracePeriodSecondsKey: "30",
+		}},
+		{secure: false, helmValues: map[string]string{
+			helmDrainListenersKey:     "false",
+			helmGracePeriodSecondsKey: "30",
+		}},
+		{secure: true, helmValues: map[string]string{
+			helmDrainListenersKey:     "false",
+			helmGracePeriodSecondsKey: "30",
+		}},
+		{secure: false, helmValues: map[string]string{
+			helmDrainListenersKey:     "false",
+			helmGracePeriodSecondsKey: "0",
+		}},
+		{secure: true, helmValues: map[string]string{
+			helmDrainListenersKey:     "false",
+			helmGracePeriodSecondsKey: "0",
+		}},
```

Comment:
This is probably too many tests in the matrix, this acceptance test is now timing out after 10 minutes locally.